File: domain/src/domain/alarm-system/factories/SecurityRuleFactory.ts
import { IntrusionRule } from '../core/IntrusionRule.js'
import { ExceedingRule } from '../core/ExceedingRule.js'
import { ObjectClass } from '../core/impl/enum/ObjectClass.js'
import { DeviceId } from '../../device/core/DeviceId.js'
import { Contact } from '../../monitoring/core/Contact.js'
import { Measure } from '../../device/core/impl/enum/Measure.js'

export interface SecurityRuleFactory {
  createIntrusionRule(
    objectClass: ObjectClass,
    securityRuleId: string,
    deviceId: DeviceId,
    creatorId: string,
    contactsToNotify: Contact[],
    description: string,
    from: Date,
    to: Date
  ): IntrusionRule

  createExceedingRule(
    min: number,
    max: number,
    measure: Measure,
    securityRuleId: string,
    deviceId: DeviceId,
    creatorId: string,
    contactsToNotify: Contact[],
    description: string,
    from: Date,
    to: Date
  ): ExceedingRule
}

File: recognition/app/domain/securityrule/factories/SecurityRuleFactory.py
import datetime
from typing import List

from app.domain.anomaly.core import ObjectClass
from app.domain.device.core import DeviceId
from app.domain.monitoring.core import Contact
from app.domain.securityrule.core import IntrusionRule


class SecurityRuleFactory:

    def create_intrusion_rule(self, device_id: DeviceId, object_classes: List[ObjectClass], contacts: List[Contact],
                              start_time: datetime, end_time: datetime) -> IntrusionRule:
        return IntrusionRule(device_id, object_classes, contacts, start_time, end_time)
